compiler/rustc_const_eval/src/const_eval/machine.rs

```diff
@@ -179,7 +179,7 @@ impl<'mir, 'tcx: 'mir> CompileTimeEvalContext<'mir, 'tcx> {
         } else if Some(def_id) == self.tcx.lang_items().panic_fmt() {
             // For panic_fmt, call const_panic_fmt instead.
             let Some(const_def_id) = self.tcx.lang_items().const_panic_fmt() else {
-                bug!("`const_panic_fmt` must be defined to call `panic_fmt` in const eval")
+                throw_unsup_format!("`const_panic_fmt` must be defined to call `panic_fmt` in const eval")
```

Comment:
Where was this triggered? I don't think this should be an error, it should stay an ICE

Reply:
You could use the `expect_lang_item` api if you want to report a better error

Reply:
This wasn't triggered, the only way to trigger this is with `no_core`. I just thought an error is better than an ICE.

Reply:
yea, but an interpreter error is the wrong kind of error. Using `self.tcx.lang_items().expect(LangItem::ConstPanicFmt)` should give a nice error, too.